### PR TITLE
deposit: adapt userEXT change in literature

### PR DESCRIPTION
**Pull request by pedrogaudencio**

### CODE REVIEW

File: inspire/modules/deposit/workflows/literature.py [hunk 1]
```diff
@@ -506,9 +506,11 @@ def restructure_ffts(fft):
         userid = deposition.user_id
         user = UserInfo(userid)
         email = user.info.get('email', '')
-        external_ids = UserEXT.query.filter_by(id_user=userid).all()
+        external_ids = UserEXT.query.filter_by(id=userid).all()
         sources = ["{0}{1}".format('inspire:uid:', userid)]
-        sources.extend(["{0}".format(e_id.id) for e_id in external_ids])
+        import ipdb; ipdb.set_trace()
```
Comment (jmartinm) [inline inspire/modules/deposit/workflows/literature.py]: I would prefer to not debug production

Comment (pedrogaudencio) [inline inspire/modules/deposit/workflows/literature.py]: Oops ^^

File: inspire/modules/deposit/workflows/literature.py [hunk 2]
```diff
@@ -506,9 +506,10 @@ def restructure_ffts(fft):
         userid = deposition.user_id
         user = UserInfo(userid)
         email = user.info.get('email', '')
-        external_ids = UserEXT.query.filter_by(id_user=userid).all()
+        external_ids = UserEXT.query.filter_by(id=userid).all()
```
Comment (jmartinm) [inline inspire/modules/deposit/workflows/literature.py]:
```python
UserEXT.query.filter_by(id_user=userid).all()
```

File: inspire/modules/deposit/workflows/literature.py [hunk 3]
```diff
@@ -506,9 +506,10 @@ def restructure_ffts(fft):
...
         sources = ["{0}{1}".format('inspire:uid:', userid)]
-        sources.extend(["{0}".format(e_id.id) for e_id in external_ids])
+        sources.extend(["{0}:{1}".format(e_id.method,
+                                        e_id.id_user) for e_id in external_ids])
```
Comment (jmartinm) [inline inspire/modules/deposit/workflows/literature.py]:
```python
sources.extend(["{0}:{1}".format(e_id.method,
                                        e_id.id) for e_id in external_ids])
```

File: inspire/modules/deposit/workflows/literature.py (hunk 2 above)
Comment (pedrogaudencio) [inline inspire/modules/deposit/workflows/literature.py]: Hum but the keys are
```
+---------+------------------+------+-----+---------+-------+
| Field   | Type             | Null | Key | Default | Extra |
+---------+------------------+------+-----+---------+-------+
| id      | varbinary(255)   | NO   | PRI | NULL    |       |
| method  | varchar(50)      | NO   | PRI | NULL    |       |
| id_user | int(15) unsigned | NO   | MUL | NULL    |       |
+---------+------------------+------+-----+---------+-------+
```
Shouldn't we query the `id` key?

We'd have something like
```
mysql> select * from userEXT;
+----+--------+------------------------+
| id | method | id_user                |
+----+--------+------------------------+
| 1  | orcid  |    1234-1234-1234-1234 |
+----+--------+------------------------+
```
Right?

Comment (jmartinm) [inline inspire/modules/deposit/workflows/literature.py]: mysql> select * from userEXT;
+---------------------+--------+---------+
| id                  | method | id_user |
```
+---------------------+--------+---------+
| 0000-0002-3598-8536 | orcid  |      87 |
+---------------------+--------+---------+
1 row in set (0.00 sec)
```

https://github.com/inveniosoftware/invenio/blob/pu/invenio/modules/accounts/models.py#L325

you can see how the Foreign Key is defined.

**Comment by jmartinm** [comment]
Merged 93bb106

**Pull request closed without merging**